pandas/core/arrays/datetimes.py

```diff
@@ -1778,3 +1872,52 @@ def _maybe_localize_point(ts, is_none, is_not_none, freq, tz):
     if is_none is None and is_not_none is not None:
         ts = ts.tz_localize(**localize_args)
     return ts
+
```

Comment:
I would move this up to the section where you the 'Construction Helpers', e.g. the maybe_* routines. (and clarify that name of the section, possibly calling it Validation and Inference )

Reply:
sure.  there's also a one func in datetimelike that is only used by PeriodArray I'd like to move.  might as well do that now